src/Controls/src/Xaml/MarkupExtensions/BindingExtension.cs

```diff
@@ -21,12 +21,21 @@ public sealed class BindingExtension : IMarkupExtension<BindingBase>
 
 		BindingBase IMarkupExtension<BindingBase>.ProvideValue(IServiceProvider serviceProvider)
 		{
+			Type bindingXDataType = null;
+			if ((serviceProvider.GetService(typeof(IXamlTypeResolver)) is IXamlTypeResolver typeResolver)
+				&& (serviceProvider.GetService(typeof(IXamlDataTypeProvider)) is IXamlDataTypeProvider dataTypeProvider)
+				&& dataTypeProvider.BindingDataType != null)
+			{
+				typeResolver.TryResolve(dataTypeProvider.BindingDataType, out bindingXDataType);
+			}
```

Comment:
Could we move this block of code into the `if (TypedBinding is null)` branch? This lookup is unnecessary when the binding is compiled.